packages/nest/src/exclusive-field.validator.test.ts

```diff
@@ -0,0 +1,38 @@
+import { describe, expect, it } from '@jest/globals';
+import { Validator } from 'class-validator';
+import { ExclusiveFieldExists } from './exclusive-field.validator';
+
+class CharacterAbility {
+  strength?: number;
+  stealth?: number;
+  magic?: number;
+}
+
+class CharacterInput {
+  @ExclusiveFieldExists()
+  ability: CharacterAbility;
+}
+
+const validator = new Validator();
+
+describe('ExclusiveFieldExists', () => {
+  it('should throw zero errors if only one of the fields exist', () => {
+    expect.assertions(1);
+    const strongCharacter: CharacterInput = { ability: { strength: 10 } };
+
+    return validator.validate(strongCharacter).then((errors) => {
+      expect(errors).toHaveLength(0);
+    });
+  });
+
+  it('should return specific error message if zero fields exist', () => {
+    expect.assertions(1);
+    const invalidCharacter: CharacterInput = { ability: {} };
+
+    return validator.validate(invalidCharacter).then((errors) => {
+      expect(errors).toHaveLength(1);
+    });
+  });
```

Comment:
@CarsonF - Thanks for those examples before!  That model seems to be working, but do you have any idea why this particular test would not throw an error?  With `ability` as an empty object without any of the 3 fields I feel like it should fail.

Reply:
The input object is not of class `CharacterInput`.
https://github.com/typestack/class-validator#validating-plain-objects
https://github.com/typestack/class-transformer#plaintoclass
https://github.com/typestack/class-transformer#working-with-nested-objects